src/main/java/thecodex6824/thaumicaugmentation/api/augment/builder/impulsecannon/IImpulseCannonConversion.java

```diff
@@ -0,0 +1,123 @@
+/**
+ *  Thaumic Augmentation
+ *  Copyright (c) 2019 TheCodex6824.
+ *
+ *  This file is part of Thaumic Augmentation.
+ *
+ *  Thaumic Augmentation is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Thaumic Augmentation is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with Thaumic Augmentation.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package thecodex6824.thaumicaugmentation.api.augment.builder.impulsecannon;
+
+import net.minecraft.client.renderer.block.model.ModelResourceLocation;
+import net.minecraft.entity.EntityLivingBase;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.ResourceLocation;
+import org.jetbrains.annotations.NotNull;
+import thecodex6824.thaumicaugmentation.api.TAItems;
+import thecodex6824.thaumicaugmentation.api.augment.CapabilityAugment;
+import thecodex6824.thaumicaugmentation.api.impetus.IImpetusStorage;
+import thecodex6824.thaumicaugmentation.common.item.ItemImpulseCannonConversion;
+
+import java.util.List;
+
+public interface IImpulseCannonConversion extends IImpulseCannonAugment {
+
+    /**
+     * @return the model resource location corresponding to the cannon model when this conversion is installed.
+     * Must be registered as a variant of the cannon item using
+     * {@link net.minecraftforge.client.model.ModelLoader#registerItemVariants(Item, ResourceLocation...)}
+     */
+    default @NotNull ModelResourceLocation getLensModel() {
```

Comment:
I forget if ModelResourceLocation is a `SideOnly(CLIENT)` class or not - if it is, then this will have to either be changed or a new client-only interface has to be made. I suspect this might be why I originally had an enum instead of returning the model directly.